HP-175: 결제페이지 결제모듈 기능 및 UI 구현

minseung-gang opened the pull request:
## PR 타입

- [x] 기능 추가
- [ ] 버그 수정
- [ ] 리팩토링
- [ ] 문서 수정
- [ ] 기타

## 반영 브랜치

HP-175 → dev

## 작업 사항

- 결제 페이지 관련 기능과 UI 구현
- 주문 생성 Type,  API QueryKey 추가
- routePath에 결제 완료페이지 경로 추가
- 주문 페이지에 필요한 컴포넌트들 구현
- 주문 페이지에 사용할 locale 분리 및 관리
- 가상계좌 결제 선택 시 준비중 알림 모달 추가


## 공유 사항
<img width="613" height="125" alt="image" src="https://github.com/user-attachments/assets/cb0608a5-bba6-4b2b-b621-ea70cb499abc" />

상품 개월 수가 기획상 1, 3, 6개월인데 백엔드에서는 1~3으로 설정되어 있어 오류가 발생했습니다. 
현재 백엔드 개발자분께 확인 중이며, 해당 값이 단계별 고정인지 초기 설계인지 명확히 파악할 예정입니다.
결과에 따라 클라이언트에서 값을 조정하거나 백엔드에서 허용 범위를 변경할 계획입니다.

## 체크리스트

- [x] 빌드에 성공했나요?
- [x] 코드 컨벤션을 잘 지켰나요? (예: `cmd + opt + L`)

## 테스트 결과

 
![화면 캡처 2025-08-05 135405](https://github.com/user-attachments/assets/fe3c58f9-7391-4405-8849-4ea6b7fac902)
 <!-- 또는 실패한 테스트에 대해 설명해주세요. -->

minseung-gang committed:
fix: 상품 내부 name필드를 productName으로 수정

minseung-gang committed:
fix: 초기 렌더링 시 productData가 undefined일 경우 대비하여 item 객체에 옵셔널 체이닝 추가

minseung-gang committed:
fix: 상품 내부의 name 필드명을 productName으로 수정

minseung-gang committed:
feat: router에 결제페이지 route 추가

minseung-gang committed:
feat: index에 결제페이지 경로 추가

minseung-gang committed:
refactor: 선택된 상품만 결제에 사용되도록 로직 수정

minseung-gang committed:
feat: 주문 생성 요청 타입 정의 추가

minseung-gang committed:
feat: 주문 페이지 locale 상수 분리 및 추가

minseung-gang committed:
feat: 주문 생성 API 함수 추가

minseung-gang committed:
feat: 주문 관련 쿼리 키 추가

minseung-gang committed:
feat: 결제페이지 기능 및 결제 연동 구현

minseung-gang committed:
feat: 주문 생성 useMutation 훅 추가

minseung-gang committed:
feat: 주문 페이지에 사용될 AccordionSection 컴포넌트 추가

minseung-gang committed:
feat: 결제페이지에 사용자의 정보를 기입하는 RecipientForm 컴포넌트 추가

minseung-gang committed:
feat: 결제페이지에 결제방법을 선택하는 PaymentMethodSelector 컴포넌트 추가

minseung-gang committed:
feat: orderSummary 컴포넌트에 사용될 locale 상수 추가

minseung-gang committed:
style: input outline 스타일 수정

minseung-gang committed:
feat: email input에 autocomplete 옵션 추가

minseung-gang committed:
feat: 가상계좌 결제 선택 시 '준비중' 알림 모달 추가

minseung-gang committed:
feat: 결제페이지 주문상품 리스트 UI 컴포넌트 추가

minseung-gang committed:
feat: 결제페이지 결제금액 요약 UI 컴포넌트 추가

minseung-gang committed:
feat: zod, @hookform/resolvers 라이브러리 추가

minseung-gang committed:
refactor: 결제페이지 관련 path에 결제완료 경로 추가

minseung-gang committed:
feat: 결제페이지 결제 이후 페이지이동 로직 추가

minseung-gang committed:
refactor: routePath purchase 수정으로 인한 수정

minseung-gang committed:
refactor: routePath puchase 수정으로 인한 routePath 값 변경

minseung-gang committed:
refactor: 이동경로를 routePath 상수로 교체

minseung-gang committed:
refactor: 결제페이지 routePath 수정으로 인해 경로 수정

minseung-gang committed:
refactor: 결제페이지 routePath 수정으로 인해 경로 수정

jieun419 commented on src/pages/purchase/components/atoms/AccordionSection.tsx:
atoms 폴더에서 AccordionSection이라는 명칭에서 `Section`이라는 명칭이 오해의 소지가 있을 것 같다 생각합니다!
Section 대신 다른 단어?로 표현하면 어떠신가요? 
또는 atom의 의미는 작은 요소로 알고 있습니다! 어쩌면 `molecules` 폴더가 적합하지 않을까 싶어요!
(의견 제시라 불필요 하다면 넘어가셔도 됩니다!)

+) 지금 까지 atom으로 알고 있었는데 atoms군요..! 이번 브랜치에 같이 수정하겠습니다ㅠ

minseung-gang replied:
아 수정을 했었는데 잘못 올라갔나보네요 ㅠㅠ 저도 이 부분 때문에 뒤에 Section을 제거하고 올리려고 했었습니다!
폴더 관련해서는 지은님께서 말씀해주신 부분이 맞는 것 같아요!
현재 올려진 폴더 단위에서 한단위씩 뒤로 미뤄서 다시 푸시하겠습니다!

jieun419 commented on src/pages/purchase/components/atoms/PaymentMethodSelector.tsx:
```diff
@@ -0,0 +1,52 @@
+      key: 'CARD',
+      label: LOCALE_LABELS[locale].paymentMethod.CARD,
+    },
+    { key: 'VIRTUAL_ACCOUNT', label: LOCALE_LABELS[locale].paymentMethod.VIRTUAL_ACCOUNT },
```
문자열 CARD, VIRTUAL_ACCOUNT 등의 데이터는 추후 상수로 관리하면 좋을 것 같아요! 
(백엔드 데이터 수정 시 같이 반영해도 좋을 듯 싶습니다!)

minseung-gang replied:
네! 저도 지은님과 같은 생각입니다! 피드백 감사합니다
[refactor: CARD와 VIRTUAL_ACCOUNT를 상수로 관리 및 적용](https://github.com/happy-pill/happyPill_Front/pull/24/commits/e32ff0b2620b1b27d0569581f7531b40f3c357a4)

jieun419 commented on src/pages/purchase/components/atoms/PaymentMethodSelector.tsx:
```diff
@@ -0,0 +1,52 @@
+      <AccordionSection title={LOCALE_LABELS[locale].paymentMethod.title}>
+        <div className='grid grid-cols-4 gap-2 p-5'>
+          {PAYMENT_METHOD.map((method, idx) => (
+            <Button
```
Button 컴포넌트 대신 `StyledButton` 컴포넌트로 반영하셔야 할 것 같습니다!

jieun419 commented on src/pages/purchase/components/molecules/OrderSummary.tsx:
```diff
@@ -0,0 +1,58 @@
+              <div key={`${item.productName} + ${idx}`} className={cn('p-5', idx !== 0 && 'pt-3')}>
+                <div className={cn(`flex border-b border-[#e9e9e9] py-5`, isLast && 'border-none')}>
+                  <div className='mr-5 h-[90px] w-[90px]'>
+                    <a href={routePath.common.product.route(item.productId)}>
```
`a` 태그 대신 `Link` 태그 추천드립니다!

minseung-gang replied:
아! 제가 태그를 하나 빼먹었군용.. 감사합니다
[refactor: OrderSumaary 컴포넌트 리팩토링](https://github.com/happy-pill/happyPill_Front/pull/24/commits/7744d4cac8bc41bf413fbf50f2a277b9453852ec)

jieun419 commented on src/pages/purchase/components/molecules/RecipientForm.tsx:
```diff
@@ -0,0 +1,71 @@
+}
+
+const RecipientForm = ({ register }: RecipientFormProps) => {
+  const phoneNumbers = ['010', '011', '016', '017', '018', '019'];
```
phoneNumbers 변하지 않는 상수 데이터로 대문자로 작성하시는건 어떠신가요?

minseung-gang replied:
좋은 의견입니다 감사합니다 지은님!
[refactor: RecipientForm 컴포넌트 리팩토링](https://github.com/happy-pill/happyPill_Front/pull/24/commits/1c96f974c3e79dffe4a6daa64848eff7c3123cb4)

jieun419 commented on src/pages/purchase/index.tsx:
```diff
@@ -0,0 +1,170 @@
+export type PurchaseFormData = z.infer<typeof userSchema>;
+
+const Index = () => {
+  const [payMethod, setPayMethod] = useState<'CARD' | 'VIRTUAL_ACCOUNT'>('CARD');
```
이전 코멘트와 동일합니다!
CARD, VIRTUAL_ACCOUNT 상수로 관리하면 좋을 듯 싶어요!

jieun419 commented on src/pages/purchase/index.tsx:
```diff
@@ -0,0 +1,170 @@
+      name: '',
+      phonePrefix: '010',
+      phoneMiddle: undefined,
+      phoneLast: undefined,
```
undefined는 정의되지 않았단 의미이고 null은 명시적으로 비어있다는 의미로 
초기 값으로 `null`이 알맞을 것 같은데 어떻게 생각하시나요?

혹은 name 처럼 빈 문자열을 줘도 좋다 생각해요!

minseung-gang replied:
네 지은님! 저도 의미적으론 null이 맞다고 생각합니다.
하지만 기술적으로 제약이 있습니다. 
기본 값으로 null을 넣어주기 위해서는 zod 스키마에 null을 허용하도록 해야하는데,
```
phoneMiddle: z.string().nullable().min(1, '휴대폰 번호를 입력해주세요')
```
이렇게 nullable()을 사용하면 null 값일 때 min(1) 검증을 건너뛰어서 필수 입력 필드임에도 validation을 통과하는 문제가 생깁니다.
그래서 **빈문자열**로 수정하려고 생각을 해봤습니다!
[refactor: 결제 페이지 리팩토링](https://github.com/happy-pill/happyPill_Front/pull/24/commits/8014f0c5faa1bcad1c389942991c272845a53663)

jieun419 reviewed (commented):
고생하셨습니다! 속도가 너무 빠르세요ㅠ

- 폴더 구조에 대해 atoms는 작은 요소의 ui, 
- molecules은 atoms들이 합쳐진 덩어리,
- organism는 molecules이 구성되어 있는 section 단위로 이해하고 있습니다!

현재 컴포넌트들이 앞 단계로 하나씩 폴더 이동 해야하지 않을까 싶어서요.
혹시 제가 잘못 이해 했다면 말씀 부탁드립니다!

minseung-gang committed:
chore: atoms → molecules 폴더로 이동

minseung-gang committed:
refactor: CARD와 VIRTUAL_ACCOUNT를 상수로 관리 및 적용

minseung-gang committed:
refactor: OrderSumaary 컴포넌트 리팩토링
- a태그를 Link 태그로 수정
- Accordion 컴포넌트 경로 수정

minseung-gang committed:
refactor: RecipientForm 컴포넌트 리팩토링
- Accordian 컴포넌트 네이밍 및 경로 수정
- phoneNumbers 상수를 대문자로 수정

minseung-gang committed:
refactor: 결제 페이지 리팩토링
- 내부 컴포넌트 폴더 경로 수정
- 'CARD' 및 'VIRTUAL_ACCOUNT' 상수 분리 및 관리
- phone 관련 기본 값 string으로 수정

minseung-gang committed:
refactor: PaymentMethodSelector 컴포넌트 리팩토링
- Accordian 컴포넌트 네이밍 수정 및 경로 수정

minseung-gang committed:
refactor: Accordion 컴포넌트 네이밍 수정

minseung-gang committed:
feat: Payment_Methods 관련 타입 추가

minseung-gang commented:
좋은 피드백 감사드립니다 지은님!
폴더 관련해서는 지은님께서 하신 말씀이 맞다고 생각합니다 ㅎㅎ 그래서 폴더를 한단위씩 내려서 수정 후 푸시했습니다!
